### PR TITLE
Update alpaca-trading extension

### PR DESCRIPTION
**Pull request by jfkisafk**
## Description

- Changed toast functionalities for closing orders and positions
- Minor refactoring for file structure

## Screencast

Screenshot 2024-05-29 at 1 25 29 AM

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

**Comment by jfkisafk** [comment]
@raycastbot help

**Comment by jfkisafk** [comment]
Asking [contributors](https://www.raycast.com/srikirank) to review as well. Not sure why @raycastbot did not process that.

**Comment by pernielsentikaer** [comment]
It's because you're the author of the extension 🙂

**Comment by pernielsentikaer** [comment]
Could you give it a spin @srikirank to help @jfkisafk test it 🙂

File: extensions/alpaca-trading/src/api/orders.ts [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+}
+
+export async function cancelOrder(order: Order) {
+  const { symbol, id } = order;
+  const toast = await getExtendedToast({ style: Toast.Style.Animated, title: `❗Cancelling order for ${symbol}`, message: `Cancelling order: ${id}` });
+  const response = await fetch(`${endpoint}/orders/${id}`, { method: 'DELETE', headers });
+  if (!response.ok) {
+    captureException(new Error(`Failed to cancel order for ${symbol}: ${response.status} ${response.statusText}`));
+    toast.style = Toast.Style.Failure;
+    toast.title = `❌ Failed to close position for ${symbol}`;
```
Comment (srikirank) [inline extensions/alpaca-trading/src/api/orders.ts:49]: Minor: The order may not have been to close the position. How about `❌ Failed to cancel order for ${symbol}`?

File: extensions/alpaca-trading/src/components/closePositionForm.tsx [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    initialValues: { kind: 'all' },
+    validation: {
+      kind: FormValidation.Required,
+      quantity: value => {
+        if (values.kind === 'qty' && Number.isNaN(Number(value)) && value!.split('.')[1]?.length > 9) {
+          return 'quantity should be a number with at most 9 decimal places';
+        }
+      },
+      percentage: value => {
+        if (values.kind === 'pc' && Number.isNaN(Number(value)) && value!.split('.')[1]?.length > 9 && (Number(value) < 0 || Number(value) > 100)) {
```
Comment (srikirank) [inline extensions/alpaca-trading/src/components/closePositionForm.tsx:48]: Should the value be strictly positive?

File: extensions/alpaca-trading/src/components/closePositionForm.tsx [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+              setTimeout(revalidate, 1000);
+            }),
+        },
+        dismissAction: { title: 'No', style: Alert.ActionStyle.Cancel },
+      }),
+    initialValues: { kind: 'all' },
+    validation: {
+      kind: FormValidation.Required,
+      quantity: value => {
+        if (values.kind === 'qty' && Number.isNaN(Number(value)) && value!.split('.')[1]?.length > 9) {
```
Comment (srikirank) [inline extensions/alpaca-trading/src/components/closePositionForm.tsx:43]: Should the value be strictly positive?

Comment (jfkisafk) [reply extensions/alpaca-trading/src/components/closePositionForm.tsx:43]: Good catch, will resolve in a follow up commit.

File: extensions/alpaca-trading/src/components/closePositionForm.tsx (hunk 2 above)
Comment (jfkisafk) [reply extensions/alpaca-trading/src/components/closePositionForm.tsx:48]: Yes but the condition already checked for > 0 and < 100. Refactored to make it clearer.

File: extensions/alpaca-trading/src/api/orders.ts (hunk 1 above)
Comment (jfkisafk) [reply extensions/alpaca-trading/src/api/orders.ts:49]: Good catch. Fixed in the new commit

**Comment by jfkisafk** [comment]
@srikirank addressed your feedback in the new commit. Please take a look.

**Review by srikirank — APPROVED** [review]

**Comment by jfkisafk** [comment]
@pernielsentikaer We have @srikirank 's approval. Can we get your blessing as well?

**Review by pernielsentikaer — COMMENTED** [review]
Hi 👋 

Looks good to me, approved 🔥

**Review by pernielsentikaer — APPROVED** [review]
Hi 👋 

Looks good to me, approved 🔥

**Comment by jfkisafk** [comment]
Hello @pernielsentikaer will you merge the changes or will a bot handle that? The changelog uses PR_MERGE_DATE which was not replaced in my previous PR

**Comment by pernielsentikaer** [comment]
You made the branch protected, so the bot can't change anything - you can see it here: https://github.com/raycast/extensions/actions/runs/9381821103/job/25831887818

Can you change that?

**Comment by jfkisafk** [comment]
@pernielsentikaer sorry I had protection rules on my branches, can you please retry merging?

**Pull request merged**